### DISCUSSION TITLE
Write script to toggle other device?

### DISCUSSION THREAD

**Discussion by stephanvierkant** [zwave-js/zwave-js-ui#2883] [Q&A] [answered]
I've got a Z-wave device that doesn't support target endpoints, so I can't use the associate group to toggle another device. 

Is it possible to write a script that gets an event from one node and sends a command to another?

I'm thinking of something like this (pseudo-code):

```
const button = driver.controller.nodes.get(63);
const light = driver.controller.nodes.get(69);

button.on('event', function (event) {
    light.getBinarySwitch(2).value = event.value
}
```

Is this even possible? It feels like a valid use case, but I can't find that much information about writing/storing/executing scripts.

**Comment by robertsLando** [comment, marked as answer]
Hi @stephanvierkant! 

Yes it's possible, also now with [scheduled-jobs](https://zwave-js.github.io/zwave-js-ui/#/usage/setup?id=scheduled-jobs) it would be even easier :) 

You should bind to node `value updated` or `value notification` event:

```js
const button = driver.controller.nodes.get(63);
const light = driver.controller.nodes.get(69);

button.on('value updated', function (zwaveNode, args) {
    if(// check the valueId of the event) {
		light.setValue(valueId, value, options)
	}
}
```

**Reply by stephanvierkant** [reply]
Thanks! And where would I save this file? And why a scheduled job? This is event handler and should run permanently, not on a hourly basis?

**Reply by robertsLando** [reply]
Because otherwise you would have to manually run it on each restart. BTW right now as you said jobs would run on specific cron expressions but I plan to also add an option to run them only on startup so that would be choice in your case

**Reply by stephanvierkant** [reply]
And where would I save this file?

**Reply by robertsLando** [reply]
You don't have to save the file, just create a new scheduled job with that function

**Reply by robertsLando** [reply]
8.7.0 available now

**Reply by stephanvierkant** [reply]
But where do I put that code? In the UI?

**Reply by robertsLando** [reply]
Yes, if you check [docs](https://zwave-js.github.io/zwave-js-ui/#/usage/setup?id=scheduled-jobs) it's explained well

**Reply by stephanvierkant** [reply]
Sorry, didn't understand that I needed to update to 8.7.0 first! Thanks for helping!
